scipy/special/basic.py

```diff
@@ -2259,15 +2288,41 @@ def factorial(n, exact=False):
     >>> arr = np.array([3, 4, 5])
     >>> factorial(arr, exact=False)
     array([   6.,   24.,  120.])
+    >>> factorial(arr, exact=True)
+    array([  6,  24, 120])
     >>> factorial(5, exact=True)
     120L
 
     """
     if exact:
-        return math.factorial(n)
+        if np.ndim(n) == 0:
+            return math.factorial(n)
+        else:
+            n = asarray(n)
+            un = np.unique(n)
+
+            if any(abs(un.astype(int)) != un):
+                raise ValueError("Input must be non-negative integers for "
+                                 "'exact' mode.")
```

Comment:
well that was already changed in https://github.com/scipy/scipy/pull/5693#issuecomment-203743919  but I'll make it behave however you want for invalid input